Mathlib/CategoryTheory/Limits/FunctorCategory.lean

```diff
@@ -53,150 +53,156 @@ it suffices to show that each evaluation cone is a limit. In other words, to pro
 #align category_theory.limits.combined_is_colimit CategoryTheory.Limits.combinedIsColimit
 
 noncomputable section
 
-instance functorCategoryHasLimitsOfShape [HasLimitsOfShape J C] : HasLimitsOfShape J (K ⥤ C)
-    where HasLimit F :=
+instance functorCategoryHasLimitsOfShape [HasLimitsOfShape J C] : HasLimitsOfShape J (K ⥤ C) where 
+  has_limit F :=
     HasLimit.mk
-      { Cone := combineCones F fun k => getLimitCone _
-        IsLimit := combinedIsLimit _ _ }
+      { cone := combineCones F fun _ => getLimitCone _
+        isLimit := combinedIsLimit _ _ }
 #align category_theory.limits.functor_category_has_limits_of_shape CategoryTheory.Limits.functorCategoryHasLimitsOfShape
 
 instance functorCategoryHasColimitsOfShape [HasColimitsOfShape J C] : HasColimitsOfShape J (K ⥤ C)
-    where HasColimit F :=
+    where 
+  has_colimit _ :=
     HasColimit.mk
-      { Cocone := combineCocones _ fun k => getColimitCocone _
-        IsColimit := combinedIsColimit _ _ }
+      { cocone := combineCocones _ fun _ => getColimitCocone _
+        isColimit := combinedIsColimit _ _ }
 #align category_theory.limits.functor_category_has_colimits_of_shape CategoryTheory.Limits.functorCategoryHasColimitsOfShape
 
-instance functorCategoryHasLimitsOfSize [HasLimitsOfSize.{v₁, u₁} C] :
-    HasLimitsOfSize.{v₁, u₁} (K ⥤ C) :=
-  ⟨inferInstance⟩
+-- Porting note: previously Lean could see through the binders and infer_instance sufficed
+instance functorCategoryHasLimitsOfSize [HasLimitsOfSize.{v₁, u₁} C] : 
+    HasLimitsOfSize.{v₁, u₁} (K ⥤ C) where  
+  has_limits_of_shape := fun _ _ => inferInstance 
 #align category_theory.limits.functor_category_has_limits_of_size CategoryTheory.Limits.functorCategoryHasLimitsOfSize
 
+-- Porting note: previously Lean could see through the binders and infer_instance sufficed
 instance functorCategoryHasColimitsOfSize [HasColimitsOfSize.{v₁, u₁} C] :
-    HasColimitsOfSize.{v₁, u₁} (K ⥤ C) :=
-  ⟨inferInstance⟩
+    HasColimitsOfSize.{v₁, u₁} (K ⥤ C) where 
+  has_colimits_of_shape := fun _ _ => inferInstance 
 #align category_theory.limits.functor_category_has_colimits_of_size CategoryTheory.Limits.functorCategoryHasColimitsOfSize
 
 instance evaluationPreservesLimitsOfShape [HasLimitsOfShape J C] (k : K) :
-    PreservesLimitsOfShape J ((evaluation K C).obj k)
-    where PreservesLimit F :=
-    preservesLimitOfPreservesLimitCone (combinedIsLimit _ _) <|
-      IsLimit.ofIsoLimit (limit.isLimit _) (evaluateCombinedCones F _ k).symm
+    PreservesLimitsOfShape J ((evaluation K C).obj k) where 
+  preservesLimit {F} := by 
+    let X : (k:K)  → LimitCone (Prefunctor.obj (Functor.flip F).toPrefunctor k) := 
+    -- Porting note: now needed X has a hint
+      fun k => getLimitCone (Prefunctor.obj (Functor.flip F).toPrefunctor k)
+    refine preservesLimitOfPreservesLimitCone (combinedIsLimit _ _) <|
+      IsLimit.ofIsoLimit (limit.isLimit _) (evaluateCombinedCones F X k).symm
```

Comment:
```suggestion
    -- Porting note: added `let` statement, because `X` could not be inferred
    let X : (k:K)  → LimitCone (Prefunctor.obj (Functor.flip F).toPrefunctor k) := 
      fun k => getLimitCone (Prefunctor.obj (Functor.flip F).toPrefunctor k)
    exact preservesLimitOfPreservesLimitCone (combinedIsLimit _ _) <|
      IsLimit.ofIsoLimit (limit.isLimit _) (evaluateCombinedCones F X k).symm
```